code/modules/mob/mob_helpers.dm

```diff
@@ -22,6 +22,27 @@
 			return 0
 	return 1
 
+/mob/proc/get_screen_colour()
+
+/mob/dead/observer/get_screen_colour()
+	return list()

```

Comment:
Nope, needs it otherwise observers will be stuck with the exact same colour as they saw when they were alive. I've tested it by ghosting out (aghost and actual ghost) and returning to body and I got my colourblindness back when I re-entered